include/RAJA/ListSegment.hxx

```diff
@@ -55,9 +55,9 @@
 
 #include "RAJA/config.hxx"
 
-#include "RAJA/BaseSegment.hxx"
+#include "RAJA/internal/BaseSegment.hxx"
```

Comment:
Currently, BaseSegment needs to be seen by users if they want to create their own segment type. But, we won't need this when the IndexSet rework is done.  So I guess it's OK to put it in internal.